aten/src/ATen/core/TensorMethods.h

```diff
@@ -1299,6 +1299,10 @@ inline bool is_sparse(Tensor self) {
   return self.is_sparse();
 }
 
+inline bool Tensor::is_xla() const {
```

Comment:
I'd rather not proliferate these changes as we add device types.  They are confusing, because it's not obvious whether you are referring to device types or backends, which is the source of the bug below.  We have `is_cuda` because that existed way-back-when in python, but all the other ones are a mistake IMO.

I'd suggest just using the TensorTypeId for now.